dependencies/secret_service.go

```diff
@@ -0,0 +1,10 @@
+package dependencies
+
+import (
+	"context"
+)
+
+type SecretService interface {
+	// LoadSecret retrieves the secret value v found at key k for organization orgID.
+	LoadSecret(ctx context.Context, k string) (string, error)
```

Comment:
Should the organization id be included in this or are we going to rely on the context?

I think the latter is ok (to rely on the context) as we have pretty few references to organizations within flux itself and I kind of like that. I feel it's a good abstraction and the organization implementation is "context" to the query. But, if that is the case, then we shouldn't reference organizations in the doc comment as there is no way to get or set the organization from flux code.

One further comment, should this return `values.Value`? I asked about this in standup and was under the impression that secrets would be an object type. It seems to me like it would be useful if this returned a `values.Value` so it was up to the implementation what type it returned.

Reply:
> Should the organization id be included in this or are we going to rely on the context?

It is part of the context, the Flux side of things will not know about organizations. Agreed we should update the comments that reference them.

> One further comment, should this return values.Value? I

No, this is the lower level API for retrieving the secret value itself. The `secrets.get` Flux code will not use this API directly, rather that will be the responsibility of the `http.post` function for example to use this service.

@aanthony1243 
That brings up a question for me though: How does `secrets.get` uses this service to check for authorization? Does it just load the secret and then discard the value and check the error?

Reply:
@nathanielc I believe that  we can use the permissions API on the influxdb side to pre-auth without loading the secret.  But in this design I had presumed we would just pull the secret and discard it.